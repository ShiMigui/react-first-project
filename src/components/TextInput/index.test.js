import { render, screen } from "@testing-library/react";
import { TextInput } from '.';

const mock = {
    placeholder: 'type value',
    onChange: jest.fn(),
    type: 'search',
    value: '',
}
const searchValue = 'test value';
const element = <TextInput {...mock} />;

describe('<TextInput />', () => {
    it('must call have a search value', () => {
        render(element);

        const inp = screen.getByPlaceholderText(mock.placeholder);
        expect(inp).toBeInTheDocument();
        expect(inp.value).toBe(mock.value);
    })

    it(`must call onChange function ${searchValue.length} times`, () => {
        render(element);

        const inp = screen.getByPlaceholderText(mock.placeholder);

        searchValue.split('').forEach(l => {
            inp.value += l;
            mock.onChange();
        });

        expect(inp.value).toBe(searchValue);
        expect(mock.onChange).toHaveBeenCalledTimes(searchValue.length);
    })

    it("must be equals to it's snapshot", () => {
        render(element);
        expect(screen.getByPlaceholderText(mock.placeholder)).toMatchSnapshot();
    })
})