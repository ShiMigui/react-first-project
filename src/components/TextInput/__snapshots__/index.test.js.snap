// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> must be equals to it's snapshot 1`] = `
<input
  class="text-input"
  placeholder="type value"
  type="search"
  value=""
/>
`;
